pkgs/desktops/lomiri/data/lomiri-session/default.nix

```diff
@@ -1,153 +1,40 @@
-      hash = "sha256-qtArOG4gysFWGnXbz3KpXEppaZ1PGDQKEGqnJvU6/RE=";
-    })
-    (fetchpatch {
-      name = "0106-lomiri-session-Change-envvar-for-1-time-binary.patch";
-      url = "https://gitlab.com/ubports/development/core/lomiri-session/-/commit/0cd1dbd30f3d5c6e50bce79146e8511e0ee56153.patch";
-      hash = "sha256-b8/Mrs36JPJE6l6/Dc/PN+zNV8Oq37HOFx+zMQvWPBY=";
-    })
-    (fetchpatch {
-      name = "0107-lomiri-session-Drag-lomiri-process-under-umbrella-of-wrapper-script.patch";
-      url = "https://gitlab.com/ubports/development/core/lomiri-session/-/commit/d8212b1862924eb283fd1ee7ea390a144d5ee97e.patch";
-      hash = "sha256-UJzV0pYEBBrXSpYxdFoBoMRzPeIQtvtPzDW2/Ljz+uI=";
-    })
-    (fetchpatch {
-      name = "0108-lomiri-session-Dont-hide-exit-code-from-systemd.patch";
-      url = "https://gitlab.com/ubports/development/core/lomiri-session/-/commit/9ac78d736caa891a4923be8d040fe8224e335215.patch";
-      hash = "sha256-yPg1K0IfaGYKqg9536i9AFCLTcAENlsJNdHjrElSeZ4=";
-    })
-
-    # Don't require a C & C++ compiler, nothing to compile
-    # Remove when https://gitlab.com/ubports/development/core/lomiri-session/-/merge_requests/14 merged & in release
-    (fetchpatch {
-      name = "0200-lomiri-session-Dont-require-a-compiler.patch";
-      url = "https://gitlab.com/ubports/development/core/lomiri-session/-/commit/6915a5678e09e5dbcb26d29a8e5585d032a19186.patch";
-      hash = "sha256-2SWiOLDLsdTjRHaJcnZe/WKcFMFmHtpZsuj7bQCtB4A=";
-    })
-
-    # Use GNUInstallDirs for install locations, find_program() for locations of used binaries
-    # fetchpatch2 due to renames, need to resolve merge conflict manually in postPatch
-    # Remove when https://gitlab.com/ubports/development/core/lomiri-session/-/merge_requests/15 merged & in release
-    (fetchpatch2 {
-      name = "0201-lomiri-session-Hardcode-less-locations.patch";
-      url = "https://gitlab.com/ubports/development/core/lomiri-session/-/commit/d5b93ecaf08ba776a79c69e8a9dd05d0b6181947.patch";
-      excludes = [ "systemd/lomiri.service" ];
-      hash = "sha256-BICb6ZwU/sUBzmM4udsOndIgw1A03I/UEG000YvMZ9Y=";
-    })
-
-    ./1001-Unset-QT_QPA_PLATFORMTHEME.patch
-  ];
+  patches = [ ./1001-Unset-QT_QPA_PLATFORMTHEME.patch ];
 
   postPatch = ''
-    # Resolving merge conflict
-    mv systemd/lomiri.service{,.in}
-    substituteInPlace systemd/lomiri.service.in \
-      --replace-fail '/usr/bin/lomiri-session' '@CMAKE_INSTALL_FULL_BINDIR@/lomiri-session' \
-      --replace-fail '/usr/bin/dbus-update-activation-environment' '@DUAE_BIN@'
-
-    substituteInPlace lomiri-session \
+    substituteInPlace lomiri-session.in \
       --replace-fail '/usr/libexec/Xwayland.lomiri' '${lib.getBin lomiri}/libexec/Xwayland.lomiri'
+
+    substituteInPlace systemd/CMakeLists.txt \
+      --replace-fail 'pkg_get_variable(SYSTEMD_USER_DIR systemd systemduserunitdir)' 'pkg_get_variable(SYSTEMD_USER_DIR systemd systemduserunitdir DEFINE_VARIABLES prefix=''${CMAKE_INSTALL_PREFIX})'
```

Comment:
Set environment variables for pkgconfig should be the simpler way

https://github.com/NixOS/nixpkgs/blob/a1521bc2d16d34067ee45aac75387eaa63f638c0/pkgs/desktops/deepin/core/dde-shell/default.nix#L74

Reply:
That's simpler, but it circumvents the whole pkg-config process abit too much for my taste tbh.